cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -60,7 +67,11 @@ protected void doResume() throws Exception {
   @Override
   protected void doStop() throws Exception {
     if (service.state() != Service.State.TERMINATED && service.state() != Service.State.FAILED) {
+      LOG.debug("stopping service for program {}.", getProgramId());
       service.stopAndWait();
+      LOG.debug("stopped service for program {}, waiting for it to finish running listener hooks.", getProgramId());
```

Comment:
this will have conflicts with:
https://github.com/caskdata/cdap/pull/7323
I renamed the method `getProgramRunId` or something like that, to be more detailed (also include runid).